### PR TITLE
Added ability to modify hostname

### PR DESCRIPTION
**Pull request by lukepayne**
Kernel: Added the ability to set the hostname via new syscall
Userland/hostname: Now takes parameter to set the hostname
LibC/unistd: Added sethostname function

### CODE REVIEW

**Comment by awesomekling** [comment]
Coolio :)

File: Kernel/Process.cpp [hunk 1]
```diff
@@ -712,6 +712,18 @@ int Process::sys$gethostname(char* buffer, ssize_t size)
     return 0;
 }
 
+int Process::sys$sethostname(const char* hostname, ssize_t length)
+{
+    REQUIRE_PROMISE(stdio);
+    if (length < 0)
+        return -EINVAL;
```
Comment (bugaevc) [inline Kernel/Process.cpp:719]: So what, any random process on the system can just go ahead and change hostname?

File: Kernel/Process.cpp [hunk 2]
```diff
@@ -712,6 +712,18 @@ int Process::sys$gethostname(char* buffer, ssize_t size)
...
+    LOCKER(*s_hostname_lock, Lock::Mode::Shared);
```
Comment (bugaevc) [inline Kernel/Process.cpp:720]: Clearly, this cannot take a shred lock.

File: Userland/hostname.cpp [hunk 3]
```diff
@@ -36,14 +36,24 @@ int main(int argc, char** argv)
...
     }
 
-    (void)argc;
-    (void)argv;
-    char buffer[HOST_NAME_MAX];
-    int rc = gethostname(buffer, sizeof(buffer));
-    if (rc < 0) {
-        printf("gethostname() error: %s\n", strerror(errno));
+    if (argc == 1) {
+      char buffer[HOST_NAME_MAX];
```
Comment (bugaevc) [inline Userland/hostname.cpp:40]: Don't do that, use `Core::ArgsParser`

File: Kernel/Process.cpp (hunk 1 above)
Comment (awesomekling) [reply Kernel/Process.cpp:719]: My bad.

File: Kernel/Process.cpp (hunk 2 above)
Comment (awesomekling) [reply Kernel/Process.cpp:720]: My bad.

**Pull request merged**